python/server/qgsserviceregistry.sip

```diff
@@ -0,0 +1,83 @@
+/***************************************************************************
+                          qgsserviceregistry.h
+
+  Class defining the service manager for QGIS server services.
+  -------------------
+  begin                : 2016-12-05
+  copyright            : (C) 2016 by David Marteau
+  email                : david dot marteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+/**
+ * \ingroup server
+ * QgsServiceRegistry
+ * Class defining the cegistry manager for QGIS server services
+ *
+ * This class provides methods for registering and retrieving 
+ * services. Note that the regstiry does not hord ownership of
+ * registered service but vill call the 'release' method on cleanup)
+ *
+ */
+class QgsServiceRegistry
+{
+%TypeHeaderCode
+#include "qgsserviceregistry.h"
+#include "qgsservice.h"
+#include "qgsserverrequest.h"
+#include "qgsserverresponse.h"
+%End
+  public:
+
+    //! Constructor
+    QgsServiceRegistry();
+
+    //! Destructor
+   ~QgsServiceRegistry();
+
+    /**
+     * Retrieve a service from its name
+     * @param name the name of the service 
+     * @param version the required version (optional)
+     * @return QgsService
+     */
+    QgsService* getService( const QString& name, const QString& version = QString() );
```

Comment:
I would very much like to leave the versioning mechanism to individual services. So there would be one WMS service that handles all WMS requests. Often the changes between protocol versions are quite small, so it is useful to have one service for multiple versions.